### PR TITLE
Avoid top segment perforation depth outside perforated grid cell. 

### PR DESCRIPTION
**Pull request by vkip**

### CODE REVIEW

**Comment by vkip** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by vkip** [comment]
jenkins build this please

File: src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp [hunk 1]
```diff
@@ -62,6 +62,8 @@
 #include <numeric>
 #include <tuple>
 
+#include <iostream>
```
Comment (GitPaean) [inline src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp:65]: We should not need this header file.

File: src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp [hunk 2]
```diff
@@ -155,7 +157,7 @@ EclipseGrid::EclipseGrid(const GridDims& gd)
 
 
 EclipseGrid::EclipseGrid(size_t nx, size_t ny , size_t nz,
-                         double dx, double dy, double dz)
+                         double dx, double dy, double dz, double top)
```
Comment (GitPaean) [inline src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp:160]: Can you give some explanation why the extra `top` argument is added here?

Comment (GitPaean) [reply src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp:160]: Thanks. It makes sense.

**Review by GitPaean — APPROVED** [review]
Thanks for the contribution. I am happy with how it looks now.  Prepared to get it in after another jenkins run.

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
The jenkins failures are due to tiny rounding errors

```
239  segment_depth 2793.4699999999997999
240  cdepth 2793.4736000000002605
```

**Comment by GitPaean** [comment]
jenkins build this update_data please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=1035 please

**Comment by GitPaean** [comment]
jenkins build this opm-tests=1035 please

**Pull request merged**